internal/services/resource/resource_group_resource.go

```diff
@@ -1,175 +1,225 @@
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
 	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
-	"github.com/hashicorp/terraform-provider-azurerm/helpers/azure"
 	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
-	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/services/resource/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/resource/validate"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/tags"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
-	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
 	"github.com/hashicorp/terraform-provider-azurerm/utils"
 )
 
-func resourceResourceGroup() *pluginsdk.Resource {
-	return &pluginsdk.Resource{
-		Create: resourceResourceGroupCreateUpdate,
-		Read:   resourceResourceGroupRead,
-		Update: resourceResourceGroupCreateUpdate,
-		Delete: resourceResourceGroupDelete,
-		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
-			_, err := parse.ResourceGroupID(id)
-			return err
-		}),
-
-		Timeouts: &pluginsdk.ResourceTimeout{
-			Create: pluginsdk.DefaultTimeout(90 * time.Minute),
-			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
-			Update: pluginsdk.DefaultTimeout(90 * time.Minute),
-			Delete: pluginsdk.DefaultTimeout(90 * time.Minute),
-		},
+type ResourceGroupResource struct{}
 
-		Schema: map[string]*pluginsdk.Schema{
-			"name": azure.SchemaResourceGroupName(),
+type ResourceGroupModel struct {
+	Name     string            `tfschema:"name"`
+	Location string            `tfschema:"location"`
+	Tags     map[string]string `tfschema:"tags"`
+}
 
-			"location": azure.SchemaLocation(),
+var _ sdk.ResourceWithUpdate = ResourceGroupResource{}
 
-			"tags": tags.Schema(),
-		},
+func (r ResourceGroupResource) ModelObject() interface{} {
+	return &ResourceGroupModel{}
+}
+
+func (r ResourceGroupResource) ResourceType() string {
+	return "azurerm_resource_group"
+}
+
+func (r ResourceGroupResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return validate.ResourceGroupID
+}
+
+func (r ResourceGroupResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": commonschema.ResourceGroupName(),
+
+		"location": commonschema.Location(),
+
+		"tags": commonschema.Tags(),
 	}
 }
 
-func resourceResourceGroupCreateUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
-	client := meta.(*clients.Client).Resource.GroupsClient
-	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
-	defer cancel()
+func (r ResourceGroupResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r ResourceGroupResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 90 * time.Minute,
```

Comment:
why does creating a RG take 90m?

Reply:
I just ported over the current values in the resource, purposely changing as little as possible in the process to avoid any behavioural difference.